sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TextTranslationClientBuilder.java

```diff
@@ -342,11 +361,21 @@ private HttpPipeline createHttpPipeline() {
         policies.add(new AddDatePolicy());
         if (tokenCredential != null) {
             policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+            if (this.region != null || this.resourceId != null) {
+                Objects.requireNonNull(this.region, "'region' cannot be null.");
+                Objects.requireNonNull(this.resourceId, "'resourceId' cannot be null.");
```

Comment:
These 2 lines are not required as the `null` checks are already done in the `if` statement.